Bottom: 6df6cace7eaf28129bd3e39b78deb3e45be0a516
Top:    7a9147bbf4814b4ce56b28a03dae65dad5aa5118
Author: Jason Wang <jasowang@redhat.com>
Date:   2018-04-17 07:41:16 +0800

Refresh of tuntap-simplify-error-handling

---

diff --git a/drivers/net/tun.c b/drivers/net/tun.c
index f906274..e180bda 100644
--- a/drivers/net/tun.c
+++ b/drivers/net/tun.c
@@ -1633,6 +1633,9 @@ static struct sk_buff *tun_build_skb(struct tun_struct *tun,
 	if (copied != len)
 		return ERR_PTR(-EFAULT);
 
+	get_page(alloc_frag->page);
+	alloc_frag->offset += buflen;
+
 	/* There's a small window that XDP may be set after the check
 	 * of xdp_prog above, this should be rare and for simplicity
 	 * we do XDP on skb in case the headroom is not enough.
@@ -1642,9 +1645,6 @@ static struct sk_buff *tun_build_skb(struct tun_struct *tun,
 	else
 		*skb_xdp = 0;
 
-	get_page(alloc_frag->page);
-	alloc_frag->offset += buflen;
-
 	preempt_disable();
 	rcu_read_lock();
 	xdp_prog = rcu_dereference(tun->xdp_prog);
